### PR TITLE
Don't send email notifications when posting delegate reports for old

### LINKED ISSUE #704: Disable email notifications for delegate reports for old competitions
@jfly said:

> The delegate reports feature is live now. One unexpected thing: delegates are going to see an alert for past competitions where they *did* submit a delegate report, but our database doesn't know about it. For example, I see this on BASC82016:
> 
> image
> 
> This might cause some confusion for delegates. A few options:
> Do nothing (easiest).
> Remove the alert entirely (pretty easy).
> Slight variant: Only show this alert for competitions that ended after June 5th.
> Create empty delegate reports for all past competitions (this is slightly nicer than #2 in that delegates will see this alert for future competitions)
> Slight variant: encourage delegates to fill in all their old competition delegate reports. This is a lot of work, but could be really nice, as it means we'll have easy access to all our past delegate reports in our database. Note: if you post a delegate report from the distant past, it will still cause an email to be sent to delegates@.

@viroulep said:

> Even though this would be a lot of work, I also think it would be quite nice to encourage Delegates to upload their previous reports! Or at least reports for "major" competitions.
Is it possible/easy to remove alerts/mailing the delegates list for reports regarding competitions before today?

So, we want to disable email notifications for delegate reports for competitions that ended before this weekend (I'm somewhat arbitrarily picking June 1st, 2016 as the cutoff).

### PR DESCRIPTION
**Pull request by jfly**
competitions. This fixes #704.

### CODE REVIEW

File: WcaOnRails/app/controllers/delegate_reports_controller.rb [hunk 1]
```diff
@@ -33,8 +33,12 @@ def update
     if @delegate_report.update_attributes(delegate_report_params)
       flash[:success] = "Updated report"
       if @delegate_report.posted? && !was_posted
-        CompetitionsMailer.notify_of_delegate_report_submission(@competition).deliver_later
-        flash[:info] = "Your report has been posted!"
+        if @competition.end_date >= Date.new(2016, 6, 1)
+          CompetitionsMailer.notify_of_delegate_report_submission(@competition).deliver_later
+          flash[:info] = "Your report has been posted and emailed!"
+        else
+          flash[:info] = "Your report has been posted but not emailed because it is for an old competition."
```
Comment (larspetrus) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: *... because it is for a pre June 2016 competition"

Comment (jfly) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: Good idea!

File: WcaOnRails/app/controllers/delegate_reports_controller.rb [hunk 2]
```diff
@@ -33,8 +33,12 @@ def update
     if @delegate_report.update_attributes(delegate_report_params)
       flash[:success] = "Updated report"
       if @delegate_report.posted? && !was_posted
-        CompetitionsMailer.notify_of_delegate_report_submission(@competition).deliver_later
-        flash[:info] = "Your report has been posted!"
+        if @competition.end_date >= Date.new(2016, 6, 1)
```
Comment (jonatanklosko) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: What about the link to #704 ?

Comment (jfly) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: WHOOPS. I created the issue just to link to it, too =)

**Comment by jonatanklosko** [comment]
LGTM!

**Pull request merged**